Add dynamic log types for functional blocks

cjdoucette opened the pull request:
This patch adds a logging type for the CPS block. I will update this pull request with log types for the other blocks after getting feedback on the setup for the CPS block.

AltraMayor commented on lua/cps.lua:
```diff
@@ -24,6 +28,7 @@ return function (net_conf, gk_conf, gt_conf, lls_conf, numa_table)
 	cps_conf.mailbox_mem_cache_size = mailbox_mem_cache_size
 	cps_conf.mailbox_burst_size = mailbox_burst_size
 	cps_conf.debug = false
```
You forgot to drop the line above.

cjdoucette commented:
Just rebased with:

- Patches for adding dynamic logging types for each block.
- A patch for removing any remaining exclamation points from log strings (for consistency)
- A patch for removing the `LUA` logtype, using either the generic `GATEKEEPER` type or the individual block's type.

As a part of the patches for the functional blocks, there are places where I changed the log type from library types (`MALLOC`, `HASH`, etc) to the functional block's type. I think it makes more sense for us to be able to control all logs for each block with a single log level instead of having some other library logging types inside of the blocks as well.

AltraMayor commented:
The pull request #160, just merged, adds three `RTE_LOG()`. Could you check if those `RTE_LOG()`s need to be changed?

cjdoucette commented:
Those calls to `RTE_LOG()` are in lib/net.c, which I have left to just use the `GATEKEEPER` static logging type. So no changes are needed.

AltraMayor commented:
The third `RTE_LOG()` of pull request #160 has an exclamation mark, so it must be edited in patch `gatekeeper: remove exclamations from logs`.

AltraMayor commented on gt/main.c:
```diff
@@ -110,8 +115,8 @@ gt_parse_incoming_pkt(struct rte_mbuf *pkt, struct gt_packet_headers *info)
-                        RTE_LOG(ERR, GATEKEEPER,
-                                "gt: failed to parse the packet's outer IPv6 extension headers!\n");
+                        GK_LOG(ERR,
+                                "Failed to parse the packet's outer IPv6 extension headers\n");
```
It should use `G[T]_LOG()` here.

cjdoucette commented:
I see, just removed and rebased.

AltraMayor commented on lua/lls.lua:
```diff
@@ -7,7 +7,10 @@ return function (net_conf, numa_table)
+
+	-- Log level for LLS. Note: set to RTE_LOG_DEBUG
+	-- to output routing table updates to the log.
+	lls_conf.log_level = gatekeeper.c.RTE_LOG_DEBUG
```
Complement the second line of the comment above as follows:
```
-- to log routing table updates and to log all the cache when it is periodically scanned.
```
Break line as needed. I used the old comment of the dropped parameter `debug` to complement the comment above.

cjdoucette commented:
Just rebased with the changes to GT and LLS.
